game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -1752,19 +1753,45 @@ private void resetCanDieFromReachingMaxDamage() {
   }
 
   private void setIsSuicide(final String s) {
-    isSuicide = getBool(s);
+    setIsSuicide(getBool(s));
   }
 
   private void setIsSuicide(final Boolean s) {
-    isSuicide = s;
+    setIsSuicideOnAttack(s);
+    setIsSuicideOnDefense(s && !Properties.getDefendingSuicideAndMunitionUnitsDoNotFire(getData()));
```

Comment:
Why is this the condition for suicide on defense?
`!Properties.getDefendingSuicideAndMunitionUnitsDoNotFire(getData()));`

Reply:
Because this is a hack property that was added to make it so all suicide units in a map wouldn't suicide on defense. This is essentially deprecated now and just used for backwards compatibility.

Reply:
Would you mind adding a comment to the code to clarify point? I'm pretty sure others would have the same question casually reading this and they would not have the benefit of this PR review conversation for reference.

Reply:
Added. Also marked these as deprecated.